### DISCUSSION TITLE
how to set the message expired and turn into dead letter exchange using pika

### DISCUSSION THREAD

**Discussion by honghh2018** [pika/pika#1525] [Q&A]
import pika # 最新版1.3.2

# producers.py
connection = pika.BlockingConnection(pika.ConnectionParameters(host='localhost', heartbeat=0))

print("connection->",connection)
channel = connection.channel()

# 2 创建队列
channel.queue_declare(queue='hello',durable=True)  # 并指定队列名称queue="name"  持久化durable=T

# 3 向指定队列插入数据
channel.basic_publish(exchange='',  # 简单模式
                      routing_key='hello',  # 指定队列,向那个队列中插入数据
                      body='Hello Yuan!'  # 写入 Hello Yuan字符串
                      )

print(" [x] Sent 'Hello Yuan!'")

consumer.py

import pika

# credentials = pika.PlainCredentials("honghh1", "123")
# connection = pika.BlockingConnection(pika.ConnectionParameters('localhost',heartbeat=0,credentials=credentials))
connection = pika.BlockingConnection(pika.ConnectionParameters('localhost',heartbeat=0))
# connection = pika.BlockingConnection(pika.ConnectionParameters('127.0.0.1',heartbeat=0))
channel = connection.channel()

#  创建队列
channel.queue_declare(queue='hello',durable=True)

# 确定回调函数  (这里只定义，其他地方调用才执行)
def callback(ch, method, properties, body):
    print(" [x] Received %r" % body)
    message=body.decode('utf-8')
    print(" [x] Received1 %s" % message)

# 确定监听队列参数#
channel.basic_consume(queue='hello',
                      auto_ack=True, # 默认应答
                      on_message_callback=callback)

print(' [*] Waiting for messages. To exit press CTRL+C')
# 正式监听
channel.start_consuming()

Any advice would be appreciated
Best,
hanhuihong

**Comment by Dheeraj4103** [comment]
@lukebakken can I work on ths ?

**Comment by lukebakken** [comment]
@Dheeraj4103 I don't know what this person is asking for. It looks like a question, so I'm going to convert this issue into a discussion. You can assist this person in the discussion.
